### PR TITLE
Fixed #31250 -- Ignored processing instructions in assertXMLEqual()/assertXMLNotEqual().

### PR DESCRIPTION
**Pull request by Irisoz**
Fix compare_xml function in assertXMLEqual and assertXMLNotEqual

### CODE REVIEW

File: django/test/utils.py [hunk 1]
```diff
@@ -576,7 +588,7 @@ def check_element(want_element, got_element):
 
     def first_node(document):
         for node in document.childNodes:
-            if node.nodeType not in (Node.COMMENT_NODE, Node.DOCUMENT_TYPE_NODE):
+            if node.nodeType != Node.COMMENT_NODE:
```
[the remark this answers is not in the document]
Comment (Irisoz) [reply django/test/utils.py]: I understood. Fixed

**Comment by claudep** [comment]
The documentation also needs an update.
https://docs.djangoproject.com/en/dev/topics/testing/tools/#django.test.SimpleTestCase.assertXMLEqual

**Comment by felixxm** [comment]
@Irisoz Thanks for updates :+1: I pushed docs changes.

**Comment by felixxm** [comment]
Merged in 54b7af7eb4107f933d14da0737b539bbe38d0fc8.

**Pull request closed without merging**